lib/core/jdl_application_configuration_factory.js

```diff
@@ -22,32 +22,27 @@ const StringJDLApplicationConfigurationOption = require('./string_jdl_applicatio
 const IntegerJDLApplicationConfigurationOption = require('./integer_jdl_application_configuration_option');
 const BooleanJDLApplicationConfigurationOption = require('./boolean_jdl_application_configuration_option');
 const ListJDLApplicationConfigurationOption = require('./list_jdl_application_configuration_option');
-const {
-  getTypeForOption,
-  doesOptionExist,
-  OptionTypes,
-  shouldTheValueBeQuoted
-} = require('./jhipster/application_options');
-
-const logger = require('../utils/objects/logger');
+const GenericJDLApplicationConfigurationOption = require('./generic_jdl_application_configuration_option');
+const { getTypeForOption, OptionTypes, shouldTheValueBeQuoted } = require('./jhipster/application_options');
 
 module.exports = { createApplicationConfigurationFromObject };
 
 function createApplicationConfigurationFromObject(configurationObject = {}) {
   const configuration = new JDLApplicationConfiguration();
   Object.keys(configurationObject).forEach(optionName => {
     const optionValue = configurationObject[optionName];
-    if (!doesOptionExist(optionName, optionValue)) {
-      logger.debug(`Unrecognized application option name and value: ${optionName} and ${optionValue}`);
-      return;
-    }
     configuration.setOption(createJDLConfigurationOption(optionName, optionValue));
   });
   return configuration;
 }
 
 function createJDLConfigurationOption(name, value) {
-  const type = getTypeForOption(name);
+  let type;
+  try {
+    type = getTypeForOption(name);
+  } catch (error) {
+    return new GenericJDLApplicationConfigurationOption(name, value);
```

Comment:
So here generic means unknown? As I understand it, `GenericJDLApplicationConfigurationOption` is a wrapper around options that have unknown types. By the way, what unknown type could there be?

Reply:
The only different behaviour is list(array/set).
Others (boolean, integer, string) can have the same behaviour.
String have a quoted predefined option. Boolean/Integer are dump.
For JCore 8 I suggest to remove every one and keep only JDLApplicationConfigurationOption with Generic behaviour.